### PR TITLE
[16.0][MIG] web_calendar_slot_duration: Migration to version 16.0

### PR DESCRIPTION
**Pull request by stefan-tecnativa**
@Tecnativa TT45649

This PR superseeds: #2601 

@chienandalu @CarlosRoca13 please review!

### CODE REVIEW

**Review by CRogos — APPROVED** [review]
Code review + function testing LGTM! 
Migration guide followed.

File: web_calendar_slot_duration/static/src/js/calendar_model.esm.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            this.env.searchModel.context.calendar_slot_duration &&
+            partialRecord.isAllDay != true
+        ) {
+            var slot_duration = this.env.searchModel.context.calendar_slot_duration;
+            var slot_duration_parts = slot_duration.match(/(\d+):(\d+):(\d+)/);
+            const hours = parseInt(slot_duration_parts[1]);
+            const minutes = parseInt(slot_duration_parts[2]);
+            const seconds = parseInt(slot_duration_parts[3]);
+            const durationFloat = hours + minutes / 60 + seconds / 3600;
+            partialRecord.end = partialRecord.start.plus({hours: durationFloat});
```
Comment (chienandalu) [inline web_calendar_slot_duration/static/src/js/calendar_model.esm.js]: suggestion: You can destructure it for better reading (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment):
```suggestion
            const slot_duration = this.env.searchModel.context.calendar_slot_duration;
            let [_s, hours, minutes, seconds] = slot_duration.match(/(\d+):(\d+):(\d+)/);
            let durationFloat = hours + minutes / 60 + seconds / 3600;
            partialRecord.end = partialRecord.start.plus({hours: durationFloat});
```

File: web_calendar_slot_duration/static/src/js/calendar_model.esm.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/** @odoo-module **/
+/* Copyright 2021 Tecnativa - Jairo Llopis
```
Comment (chienandalu) [inline web_calendar_slot_duration/static/src/js/calendar_model.esm.js]: This is completely refactored by you, isn't it?

**Review by chienandalu — CHANGES_REQUESTED** [review]
Very good work @stefan-tecnativa :smile: :+1: 

I don't think a supersed applies for this case. The previous PRs weren't aimed at all and no base work was used for this one so I don't think is fair to coauthor the migration commit.

Some other suggestions:

File: web_calendar_slot_duration/static/src/js/calendar_model.esm.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl). */
+
+import {CalendarModel} from "@web/views/calendar/calendar_model";
+import {patch} from "@web/core/utils/patch";
+
+patch(CalendarModel.prototype, "WebCalendarSlotDurationCalendarModel", {
+    buildRawRecord(partialRecord, options = {}) {
+        if (
+            this.env.searchModel.context.calendar_slot_duration &&
+            partialRecord.isAllDay != true
```
Comment (CRogos) [inline web_calendar_slot_duration/static/src/js/calendar_model.esm.js]: Could you fix the pre-commit issue?

**Review by CRogos — CHANGES_REQUESTED** [review]
I also found another issue when trying to open the edit form

[calendar_slot_duration.webm](https://github.com/OCA/web/assets/1799080/283f005e-31eb-4749-bca4-917ee6bf51fa)

```
InvalidArgumentError: Invalid unit value 00:15:0000.004166666666666667
    at asNumber (http://oca-web-16-0-pr2646-05137d2aeb50.runboat.odoo-community.org/web/assets/202-bf3d5c2/web.assets_common.min.js:872:131)
    at normalizeObject (http://oca-web-16-0-pr2646-05137d2aeb50.runboat.odoo-community.org/web/assets/202-bf3d5c2/web.assets_common.min.js:873:179)
    at Function.fromObject (http://oca-web-16-0-pr2646-05137d2aeb50.runboat.odoo-community.org/web/assets/202-bf3d5c2/web.assets_common.min.js:1011:29)
    at Function.fromDurationLike (http://oca-web-16-0-pr2646-05137d2aeb50.runboat.odoo-community.org/web/assets/202-bf3d5c2/web.assets_common.min.js:1011:403)
    at DateTime.plus (http://oca-web-16-0-pr2646-05137d2aeb50.runboat.odoo-community.org/web/assets/202-bf3d5c2/web.assets_common.min.js:1188:193)
    at AttendeeCalendarModel.buildRawRecord (http://oca-web-16-0-pr2646-05137d2aeb50.runboat.odoo-community.org/web/assets/208-35a7a33/web.assets_backend.min.js:13318:659)
    at buildRawRecord (patch WebCalendarSlotDurationCalendarModel) (http://oca-web-16-0-pr2646-05137d2aeb50.runboat.odoo-community.org/web/assets/208-35a7a33/web.assets_backend.min.js:1659:22)
    at AttendeeCalendarModel.buildRawRecord (http://oca-web-16-0-pr2646-05137d2aeb50.runboat.odoo-community.org/web/assets/208-35a7a33/web.assets_backend.min.js:12914:61)
    at AttendeeCalendarController.editRecordInCreation (http://oca-web-16-0-pr2646-05137d2aeb50.runboat.odoo-community.org/web/assets/208-35a7a33/web.assets_backend.min.js:2536:57)
    at CalendarQuickCreate.editRecord (http://oca-web-16-0-pr2646-05137d2aeb50.runboat.odoo-community.org/web/assets/208-35a7a33/web.assets_backend.min.js:2751:25)
```

**Comment by pedrobaeza** [comment]
/ocabot migration web_calendar_slot_duration

@chienandalu can you take care of this?

**Comment by chienandalu** [comment]
> @chienandalu can you take care of this?

Sure ;)

**Review by chienandalu — APPROVED** [review]
It's fixed now

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Comment by CRogos** [comment]
@chienandalu there is another issue with this module... the timespan of the created calendar entry is incorrect (always 15 min). When you uninstall the module, it is working correctly again.

[calendar_slot_duration2.webm](https://github.com/OCA/web/assets/1799080/8421dfd4-e40e-4131-9e65-39362af11daa)

**Pull request merged**